baseapp/auth/saml/helpers.go

```diff
@@ -0,0 +1,99 @@
+// Copyright 2019 Palantir Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package saml
+
+import (
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/pem"
+	"encoding/xml"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/crewjam/saml"
+	"github.com/pkg/errors"
+)
+
+func CertificateFromFile(path string) (*x509.Certificate, error) {
+	certBytes, err := ioutil.ReadFile(path)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not read provided certificate file")
+	}
+
+	return CertificateFromBytes(certBytes)
+}
+
+func CertificateFromBytes(certBytes []byte) (*x509.Certificate, error) {
+	certPem, _ := pem.Decode(certBytes)
+	if certPem == nil {
+		return nil, errors.New("could not PEM decode the provided certificate")
```

Comment:
Is the `pem.Decode` error unhelpful here?

Reply:
unusually, the second return field is actually not an error, but a byte slice of the remaining un-parsed bytes